lattice_bscan_spi.py

```diff
@@ -159,12 +160,15 @@ def elaborate(self, platform):
                 with m.If(head == 0):
                     m.next = "XFER"
             with m.State("XFER"):
+                m.d.sync += self.cs_n.o.eq(0)
                 m.d.sync += bits.eq(bits - 1)
                 with m.If(bits == 0):
-                    m.next = "IDLE"
+                    m.next = "XFER_FIN"
+            with m.State("XFER_FIN"):
+                m.d.sync += self.jtag.sel.eq(0)
```

Comment:
I believe driving `jtag.sel` by the JTAG TAP controller itself is more appropriate than our custom SPI transfer cycle FSM.